### PR TITLE
editoast: Setup `rstest` and add and use `fast_rolling_stock` fixture

### PR DESCRIPTION
**Pull request by bloussou**
This PR is:
- replacing `ntest` by `rstest`
- adding and using `db_pool` and `fast_rolling_stock` fixtures

Main point of fixtures is to be sure that setup and cleanup is clean and does not pollute test code (it is simplifying each test).

first steps of #3637

### CODE REVIEW

File: editoast/src/fixtures.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    impl<T: Delete + Id> TestFixture<T> {
+        pub fn id(&self) -> i64 {
+            self.model.get_id()
+        }
+    }
+
+    impl<T: Delete + Id> Drop for TestFixture<T> {
+        fn drop(&mut self) {
+            let _ = executor::block_on(RollingStockModel::delete(self.db_pool.clone(), self.id()));
```
Comment (flomonster) [inline editoast/src/fixtures.rs]:
```suggestion
            let _ = executor::block_on(T::delete(self.db_pool.clone(), self.id()));
```

File: editoast/src/models/mod.rs [hunk 2]
```diff
@@ -20,6 +20,10 @@ pub use study::{Study, StudyWithScenarios};
 pub use timetable::{Timetable, TimetableWithSchedules};
 pub use train_schedule::TrainSchedule;
 
+pub trait Id {
```
Comment (flomonster) [inline editoast/src/models/mod.rs]: We can rename it and implement it automatically for objects that already implement the `diesel::Identifiable<Id = i64>`

```suggestion
pub trait Identifiable {
```

```rust
impl<T: Identifiable<Id = i64> + Clone> Id for T {
    fn get_id(&self) -> i64 {
        self.clone().id()
    }
}
```

File: editoast/Cargo.toml [hunk 3]
```diff
@@ -53,4 +53,5 @@ pointy = "0.4.0"
 futures = "0.3"
 
 [dev-dependencies]
-ntest = "0.9.0"
+rstest = "0.17.0"
+async-std = { version = "1.5", features = ["attributes", "tokio1"] }
```
Comment (flomonster) [inline editoast/Cargo.toml:57]: I didn't see where you need these crates.

**Review by flomonster — CHANGES_REQUESTED** [review]
Thanks! I love this new feature :heart_eyes:.

File: editoast/Cargo.toml (hunk 3 above)
Comment (bloussou) [reply editoast/Cargo.toml:57]: We need it for async fixtures, it is specified here : https://docs.rs/rstest/latest/rstest/attr.rstest.html#async

**Review by flomonster — APPROVED** [review]
LGTM

**Pull request merged**